core/ArchiveProcessor/Loader.php

```diff
@@ -119,20 +123,77 @@ private function prepareArchiveImpl($pluginName)
         // invalidate existing archives before we start archiving in case data was tracked in the past. if the archive is
         // made invalid, we will correctly re-archive below.
         if ($this->invalidateBeforeArchiving
-            && Rules::isBrowserTriggerEnabled()
+          && Rules::isBrowserTriggerEnabled()
         ) {
             $this->invalidatedReportsIfNeeded();
         }
+        // load existing data from archive
+        list($visits, $visitsConverted) = $this->loadArchiveData();
+        if ($this->quickReturn) {
+            return [$visits, $visitsConverted];
+        }
+
+        if (SettingsServer::isArchivePhpTriggered()) {
+            $this->logger->info("initiating archiving via core:archive for " . $this->params);
+        }
+
+        //get Lock ID
+        $lockId = $this->makeArchivingLock();
 
+        //ini lock
+        $this->lock = new LoaderLock($lockId);
+
+        //set mysql lock the entire process if another process is running
+        $this->lock->setLock();
+        try {
+            //try load Archive Data
+            list($visits, $visitsConverted) = $this->loadArchiveData();
+            if ($this->quickReturn) {
+                return [$visits, $visitsConverted];
+            }
+
+            //insert data
+            list($visits, $visitsConverted) = $this->prepareCoreMetricsArchive($visits, $visitsConverted);
+            list($idArchive, $visits) = $this->prepareAllPluginsArchive($visits, $visitsConverted);
+            if ($this->isThereSomeVisits($visits) || PluginsArchiver::doesAnyPluginArchiveWithoutVisits()) {
+                return [[$idArchive], $visits];
+            }
+        } catch (\Exception $e) {
+            throw new \Exception("exceed max time");
```

Comment:
What case are you actually trying to catch here? Before there hadn't been any try catch around that code and the lock method is called outside 🤔